ezomero/_ezomero.py

```diff
@@ -329,6 +330,78 @@ def post_map_annotation(conn, object_type, object_id, kv_dict, ns,
     return map_ann.getId()
 
 
+@do_across_groups
+def post_file_annotation(conn, object_type, object_id, file_path, ns,
+                        mimetype=None, description=None, across_groups=True):
+    """Create new FileAnnotation and link to images.
+
+    Parameters
+    ----------
+    conn : ``omero.gateway.BlitzGateway`` object
+        OMERO connection.
+    object_type : str
+       OMERO object type, passed to ``BlitzGateway.getObjects``
+    object_ids : int
+        ID of object to which the new MapAnnotation will be linked.
+    file_path : string
+        local path to file to be added as FileAnnotation
+    ns : str
+        Namespace for the FileAnnotation
+    mimetype : str
+        String of the form 'type/subtype', usable for a MIME content-type header.
+    description : str
+        File description to be added to FileAnnotation
+    across_groups : bool, optional
+        Defines cross-group behavior of function - set to
+        ``False`` to disable it.
+
+    Notes
+    -----
+    All keys and values are converted to strings before saving in OMERO.
+
+    Returns
+    -------
+    file_ann_id : int
+        IDs of newly created MapAnnotation
+
+    Examples
+    --------
+    >>> ns = 'jax.org/jax/example/namespace'
+    >>> path = '/home/user/Downloads/file_ann.txt'
+    >>> post_file_annotation(conn, "Image", 56, path, ns)
+    234
+    """
+
+    if type(file_path) is not str:
+        raise TypeError('file_path must be of type `str`')
+
+    obj = None
+    if object_id is not None:
+        if type(object_id) is not int:
+            raise TypeError('object_ids must be integer')
+        obj = conn.getObject(object_type, object_id)
+        if obj is not None:
+            ret = set_group(conn, obj.getDetails().group.id.val)
+            if ret is False:
+                logging.warning('Cannot change into group '
+                                f'where object {object_id} is.')
+                return None
+        else:
+            logging.warning(f'Object {object_id} could not be found '
+                            '(check if you have permissions to it)')
+            return None
+    else:
+        raise TypeError('Object ID cannot be empty')
+    
+    mimetype = mimetypes.guess_type(file_path)
```

Comment:
wouldn't this overwrite the parameter, if it is used?

Reply:
good catch! pushed a commit fixing it